CameraOverlay/overlay_all_stats.py

```diff
@@ -1,160 +1,129 @@
-import os
 import time
-from overlay import Overlay
-from PIL import Image, ImageDraw, ImageFont
+from overlay import Overlay, Color
 import topics
 
-
 class OverlayAllStats(Overlay):
 
+	topics = [
+			topics.DAS.start,
+			topics.DAS.data,
+			topics.DAS.stop,
+			topics.PowerModel.recommended_sp,
+			topics.PowerModel.max_speed,
+	]
+
 	def __init__(self):
 		super(OverlayAllStats, self).__init__()
-
-		self.prev_time = 0
-
-		self.font_path = os.path.join(os.path.dirname(__file__), 'FreeSans.ttf')
-		self.bottom_text_height = 70
-		self.bottom_text_font = ImageFont.truetype(self.font_path, self.bottom_text_height)
-		self.top_text_height = 45
-		self.top_text_font = ImageFont.truetype(self.font_path, self.top_text_height)
-		self.top_box_height = 80
-		self.top_box_width = self.width
+		self.text_height = 50
+		self.speed_height = 70
 
 	def on_connect(self, client, userdata, flags, rc):
-		print("Connected with rc: " + str(rc))
-		# Subscribed topics
-		client.subscribe(topics.DAS.start)
-		client.subscribe(topics.DAS.data)
-		client.subscribe(topics.DAS.stop)
-		client.subscribe(topics.PowerModel.recommended_sp)
-		client.subscribe(topics.PowerModel.predicted_max_speed)
-		client.subscribe(topics.PowerModel.plan_name)
+		print('Connected with rc: {}'.format(rc))
+		for topic in OverlayAllStats.topics:
```

Comment:
if not mistaken, it's more efficient to call subscribe by passing an array of topics rather than calling mulitple subscribes

Reply:
alright, updated the code ;)